File: partition-go/partition-go/solution_v2.go
package partition_go

func PartitionV2(node *Node, data int) *Node {
	head := node
	tail := node

	for node != nil {
		next := node.Next
		if node.Data < data {
			// insert at head
			node.Next = head
			head = node
		} else {
			// insert at tail
			tail.Next = node
			tail = node
		}
		node = next
	}

	tail.Next = nil
	return head
}

File: sum-list-java/sum-list-java/src/main/java/SolutionV2.java
public class SolutionV2 {

    Node sumList(Node nodeOne, Node nodeTwo) {
        int carried = 0;
        int total = 0;

        Node previous = null;
        Node head = null;
        for (; nodeOne != null; ) {
            total = (nodeOne.data + nodeTwo.data) + carried;
            Node current = new Node(null, total % 10);

            if (total > 10){
                carried = 1;
            }else {
                carried = 0;
            }
             if (previous == null){
                previous = current;
                 head = previous;
            }else {
                previous.next = current;
                previous = current;
            }

            nodeOne = nodeOne.next;
            nodeTwo = nodeTwo.next;
        }

        return head;
    }


}